### PR TITLE
[AGENT] Improve autorecord permission diagnostics

### PR DESCRIPTION
**Pull request by BASIC-BIT**
[AGENT]

## Summary
- Require the text-channel permissions Chronote needs for meeting status and notes messages.
- Send autorecord permission failures to the configured notes channel when possible, otherwise best-effort DM the triggering user with admin-aware copy.
- Update public docs and add permission helper tests.

## Testing
- `yarn test --runTestsByPath src/utils/__tests__/permissions.test.ts --coverage=false`
- `yarn build`
- `npx eslint src/utils/permissions.ts src/commands/startMeeting.ts src/commands/autorecord.ts src/utils/__tests__/permissions.test.ts`
- `yarn docs:check`

## Risk / Rollout
- No data migration or feature flag required.
- DM delivery remains best-effort when Discord blocks bot DMs.
- PR opened as draft to avoid triggering paid review before checks are reviewed.

### CODE REVIEW

File: src/utils/__tests__/permissions.test.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+      buildAutoRecordPermissionChannelMessage({
+        voiceChannelName: "voice",
+        textChannelName: "notes",
+        missingVoicePermissions: ["Connect"],
+        missingTextPermissions: ["Read Message History", "Embed Links"],
+      }),
+    ).toBe(
+      "Cannot start auto-recording because Chronote is missing permissions in voice channel **voice**: **Connect**; notes channel **notes**: **Read Message History**, **Embed Links**.",
+    );
+  });
```
[the remark this answers is not in the document]
Comment (BASIC-BIT) [reply src/utils/__tests__/permissions.test.ts:120]: [AGENT] Fixed in `5ab7dac` by adding a regression test for the empty missing-permissions message case.

File: src/utils/permissions.ts [hunk 2]
```diff
@@ -1,10 +1,89 @@
...
+      `voice channel **${voiceChannelName}**: ${formatMissingPermissions(missingVoicePermissions)}`,
+    );
+  }
+  if (missingTextPermissions.length > 0) {
+    parts.push(
+      `notes channel **${textChannelName}**: ${formatMissingPermissions(missingTextPermissions)}`,
+    );
+  }
+  return parts.join("; ");
+};
```
[the remark this answers is not in the document]
Comment (BASIC-BIT) [reply src/utils/permissions.ts:72]: [AGENT] Fixed in `5ab7dac` by adding fallback copy for the empty missing-permissions race window.

**Comment by BASIC-BIT** [comment]
[AGENT] Addressed Greptile feedback in `5ab7dac`, replied directly to both inline threads, and resolved them.

File: src/utils/permissions.ts [hunk 3]
```diff
@@ -1,10 +1,89 @@
...
+] satisfies RequiredPermission[];
+
+const TEXT_CHANNEL_MEETING_PERMISSIONS = [
+  { flag: PermissionsBitField.Flags.ViewChannel, label: "View Channel" },
+  { flag: PermissionsBitField.Flags.SendMessages, label: "Send Messages" },
+  {
+    flag: PermissionsBitField.Flags.ReadMessageHistory,
+    label: "Read Message History",
+  },
+  { flag: PermissionsBitField.Flags.EmbedLinks, label: "Embed Links" },
```
[the remark this answers is not in the document]
Comment (BASIC-BIT) [reply src/utils/permissions.ts]: [AGENT] Fixed in `4f2a69f` by removing `Embed Links` from the required text-channel permission set, updating the permission tests, and aligning the public docs with Discord's Create Message requirement.
